### DISCUSSION TITLE
Permission Issues Accessing sdm-prod Project Resources

### DISCUSSION THREAD

**Discussion by johnhuetz** [potmat/homebridge-google-nest-sdm#148] [Q&A] [answered]
### Problem Description
I am encountering permission issues with the `homebridge-google-nest-sdm` plugin. The plugin initialization fails due to lack of access to the `sdm-prod` project. The specific errors indicate missing permissions for `pubsub.topics.get` and `serviceusage.services.list`.

### Steps to Reproduce
1. I set up the plugin according to the provided documentation, including adding the additional scope when obtaining the authorization code.
2. I confirmed that the Smart Device Management and Pub/Sub APIs are enabled.
3. I obtained the OAuth credentials and refresh token as instructed.
4. Upon running Homebridge, the plugin logs show errors related to permission issues with the `sdm-prod` project.

### Expected Behavior
The plugin should initialize successfully and connect to my Google Nest devices without requiring access to the `sdm-prod` project.

### Actual Behavior
The plugin initialization fails with the following error message:

'Plugin initialization failed, there was a failure with event subscription. StatusError: Requested project not found or user does not have access to it (project=sdm-prod)'

### Additional Context
- I have verified that my Google Cloud and Device Access Console projects are correctly linked.
- My account has Owner permissions on my Google Cloud project.
- There seems to be a misconfiguration that causes the plugin to attempt to access the 'sdm-prod' project, which I do not have permission for.

### Question
How can I configure the plugin to avoid permission issues with the 'sdm-prod' project? Is there a step I missed that grants my Google account the necessary permissions, or should the plugin reference resources only within my Google Cloud project?

**Comment by potmat** [comment, marked as answer]
Hi @johnhuetz, since many thousands of people have setup the plugin without this problem I think it is not an issue with the plugin but probably your configuration, so I've converted it to a discussion.  Remember your subscription ID field should look like "projects/your-gcp-project-id/subscriptions/your-subscription-id", if you "sdm-prod" where "your-gap-project-id" goes then you've got the wrong URL.

**Reply by johnhuetz** [reply]
Thanks, @potmat, for moving my "issue" to a discussion; I should have done that from the get-go !

I used just the subscription ID vs. the full subscription name in the config setting, a total amateur move.

I see my Nest cameras now in the apple Home app, but no video stream, but I'll keep working on it. Thanks again for the help!
